components/library/libp11/Makefile

```diff
@@ -0,0 +1,58 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 Jim Klimov
+#
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		libp11
+COMPONENT_VERSION=	0.4.2
+COMPONENT_ARCHIVE=	$(COMPONENT_NAME)-$(COMPONENT_VERSION).tar.gz
+COMPONENT_FMRI=		library/$(COMPONENT_NAME)
+COMPONENT_CLASSIFICATION=	System/Security
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_PROJECT_URL=	https://github.com/OpenSC/$(COMPONENT_NAME)/wiki
+COMPONENT_ARCHIVE_URL=	https://github.com/OpenSC/$(COMPONENT_NAME)/releases/download/$(COMPONENT_NAME)-$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+COMPONENT_ARCHIVE_HASH=	\
+	sha256:e2c3614a314b452a9b57e2914252df3ffee59e262dfb75b4fc73a2247f8ddebe
+COMPONENT_LICENSE=		LGPLv2.1+
+COMPONENT_LICENSE_FILE= COPYING
+COMPONENT_DESCRIPTION=	libp11 is a library implementing a thin layer on top of PKCS11 API to make using PKCS11 implementations easier
+COMPONENT_SUMMARY=	libp11 is a library to make using PKCS11 implementations easier
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/configure.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+COMPONENT_PREP_ACTION = \
+        (cd $(@D) &&\
+        autoreconf -fiv &&\
+        $(RM) config.h )
+
+COMPONENT_PRE_CONFIGURE_ACTION = \
+        ($(CLONEY) $(SOURCE_DIR) $(@D))
+
+INDIR.32=/usr/bin
+BINDIR.64=/usr/bin/$(MACH64)
+LIBDIR.32=/usr/lib
+LIBDIR.64=/usr/lib/$(MACH64)
+
+# Normally this expects /usr/lib/engines - but our openssl does not deliver that
```

Comment:
So, after some unraveling, it seems that:
* the `/usr/lib/libpkcs11.so*` files are delivered by `pkg:/system/library` wherever it takes the sources.
* the opessl-1.0.2 recipe builds and delivers its own (+patches) code, including the `/lib/openssl` directory with engines, some scripts, etc. and the `openssl` binary to which `/usr/bin/openssl` is just a symlink. It does not deliver a `libpkcs11.so*` although refers to it in the configure options of the build recipe.
* the libp11 makes (from different sources) a `libp11.so*` set of libraries that it installs into a common libdir (e.g. /usr/lib) and a `pkcs.so` for the engine (according to source-file set) which it puts into `--engine-dir=...` location (/usr/lib/engine by default), and makes a symlink to it as the `./libpkcs11.so` wherever it installed this piece.

I have no idea if the two variants of `libpkcs11.so` (from `system/library` in the common location and the engine from libp11) are related by source origins, or even by intention (libp11 one is very likely an openssl engine, no idea what the system one intends to be). But at least if we deliver the engine shared object into engine location, the conflict seems avoided too ;)